components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,199 @@
+		row = 'localprize'
+	},
+	[prizeTypes.POINTS] = {
+		header = 'points',
+		headerParse = function (input)
+			local pointsData = mw.loadData('Module:Points/data')
+			return pointsData[input] or {title = 'Points'}
+		end,
+		row = 'points'
+	},
+	[prizeTypes.QUALIFIES] = {
+		header = 'qualifies',
+		headerParse = function (input)
+			-- TODO: Add manual and automatic retrevial of additional data
+			return {link = input:gsub(' ', '_')}
+		end,
+		row = 'qualified'
+	},
+	[prizeTypes.FREETEXT] = {
+		header = 'freetext',
+		headerParse = function (input)
+			return {title = input}
+		end,
+		row = 'freetext'
+	},
+}
+
+function PrizePool:init(args)
+	self.args = args
+	self.pagename = mw.title.getCurrentTitle().text
+	self.args.type = Json.parseIfString(self.args.type)
+
+	if not self.args.type then
+		return error('Please provide a type!')
+	elseif type(self.args.type) ~= 'table' or not self.args.type.type then
+		return error('Could not parse type!')
+	elseif not Opponent.isType(self.args.type.type) then
+		return error('Not a valid type!')
+	end
+
+	self.type = self.args.type.type
+
+	self.options = self:_readStandardOptions(args)
+	self.prizes = self:_readStandardPrizes(args)
+	if self.options.showUSD then
+		self:addPrize('USD', 1)
+	end
+	self.placements = self:_readPlacements(args)
+end
+
+function PrizePool:create()
+	-- TODO
+	mw.logObject(self)
+	return ''
+end
+
+function PrizePool:_readStandardOptions(args)
+	self.options = {}
+
+	self:setOption('autoUSD', Logic.nilOr(Logic.readBoolOrNil(args.autousd), true))
+	self:setOption('showUSD', Logic.nilOr(Logic.readBoolOrNil(args.showusd), true)) -- TODO: Improve
+	self:setOption('prizeSummary', Logic.nilOr(Logic.readBoolOrNil(args.prizesummary), true))
+	self:setOption('exchangeInfo', Logic.nilOr(Logic.readBoolOrNil(args.exchangeinfo), true))
+	self:setOption('storeSmw', Logic.nilOr(Logic.readBoolOrNil(args.storesmw), true))
+	self:setOption('storeLpdb', Logic.nilOr(Logic.readBoolOrNil(args.storelpdb), true))
+
+	return self.options
+end
+
+function PrizePool:_readStandardPrizes(args)
+	self.prizes = {}
+
+	for prizeEnum in pairs(prizeTypes) do
+		local prizeDatum = prizeData[prizeTypes[prizeEnum]]
+		local fieldName = prizeDatum.header
+		if fieldName then
+			args[fieldName .. '1'] = args[fieldName .. '1'] or args[fieldName]
+			for _, prizeValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				local data = prizeDatum.headerParse(prizeValue)
+				self:addPrize(prizeEnum, index, data)
+			end
+		end
+	end
+
+	return self.prizes
+end
+
+function PrizePool:_readPlacements(args)
+	self.placements = {}
+
+	local currentPlace = 0
+	for placementIndex = 1, math.huge do
+		local placementInput = Json.parseIfString(args[placementIndex])
+		if not placementInput then
+			break
+		end
+
+		local placement = {}
+		-- TODO: Parse prizes
+		placement.prizes = {}
+
+		placement.opponents = {}
+		for opponentIndex = 1, math.huge do
+			if not placementInput[opponentIndex] then
+				break
```

Comment:
this kills the option to have input like this:
```
{{PrizePoolTable|points=ept2021na|points2=seed
|tournament1=ESL Pro Tour/2021/22/Masters/Fall/NA/Qualifiers#Closed Qualifier
|tournament2={{FULLPAGENAME}}
|{{PrizePoolSlot|place=1|usdprize=6950|points=400|points2=[[ESL_Pro_Tour/2021/22/Masters/Fall|Season Finals]]}}
|{{PrizePoolSlot|place=2|usdprize=3950|points=250|points2=[[ESL_Pro_Tour/2021/22/Masters/Fall|Season Finals]]}}
|{{PrizePoolSlot|place=3|usdprize=3600|points=140}}
|{{PrizePoolSlot|place=4|usdprize=2800|points=85}}
|{{PrizePoolSlot|place=5-6|usdprize=2200|points=55}}
|{{PrizePoolSlot|place=7-8|usdprize=1750|points=35}}
|{{PrizePoolSlot|place=7-8|usdprize=1650|points=35}}
|{{PrizePoolSlot|place=9-10|usdprize=1250|points=20}}
|{{PrizePoolSlot|place=11-12|usdprize=950|points=15}}
|{{PrizePoolSlot|place=11-12|usdprize=1050|points=15}}
|{{PrizePoolSlot|place=13-14|usdprize=650|points=10}}
|{{PrizePoolSlot|place=13-14|usdprize=850|points=10}}
|{{PrizePoolSlot|place=15-16|usdprize=500|points=5}}
|{{PrizePoolSlot|place=15-16|usdprize=400|points=5}}
|{{PrizePoolSlot|place=17-20|usdprize=250|count=4
|
|{{LiteralOpponent|TBD|race=t|flag=us}}
|
|
}}
|{{PrizePoolSlot|place=21-24|usdprize=150}}
}}
```
we would need to add an empty literal opponent in each of the lines
imo it would be better to do the place parsing of lines 150-153 above this loop and adjust the loop to only break if opponent is empty AND (end place - start place) is reached
in the cases of empty opponents before the end placement is reached just add an _EMPTY_OPONENT instead
that empty opponent would just inherit prize, points etc.

(in a later function call it would later get filled up by the automation)

Reply:
```
|{{PrizePoolSlot|place=17-20|usdprize=250
 |{{Opponent|}}
 |{{Opponent|}}
 |{{Opponent|}}
 |{{Opponent|}}
}}
```
Would be sufficient

Also this 
```
|{{PrizePoolSlot|place=7-8|usdprize=1750|points=35}}
|{{PrizePoolSlot|place=7-8|usdprize=1650|points=35}}
```
Is none-supported input
The expected input would be:
```
|{{PrizePoolSlot|place=7-8|usdprize=1650|points=35
 |{{Opponent|usdprize=1750|}}
 |{{Opponent|}}
}}
```

Reply:
ah overlooked the 7-8 stuff from where i copied it
still not a fan of having to input useless opponent template calls though
it just eats runtime for no value, especially if automation fills in the missing ones afterwards anyways

Reply:
Automation would sit after Template and before the invocation of Module. If it's being automated, no need to provide anything in the template code.

Reply:
well yeah you still need to enter prize values etc.
and so far we used the manual input as overwrite

Reply:
> well yeah you still need to enter prize values etc.

Meant for listing the opponents

> and so far we used the manual input as overwrite

Yes, but the manual input for the SC2 auto is up to the SC2 auto to handle.

Reply:
so we would basically ditch the commons stuff entirely?
wasn't the idea to provide such automation via commons for all
most of it is not sc2 specific but can be applied for any wiki using match2
sc2 specific is only the race and archon handling (plus setting an import limit which most other wikis just would not set)

Reply:
I think you might have missed 

> But can move the explicit numbering parse first.

(which has been done)

Reply:
just saying bracket import of dhe's version works out of the box with any match2 wiki
for group import it depends on the standings objects of GTL etc

example on R6:
https://liquipedia.net/rainbowsix/User:Hjpalpha#Prize_Pool

Reply:
Yes, but that's out of scope for this feature?

Reply:
> 1. read the wiki args
> 2. "run the automation" which would return a lua table structure understandable by the Prize Pool commons
> 3. Run the Prize Pool commons with the lua table retrieved from the auto (instead of wiki args as none-automated wikis would). Adjust it as needed by available methods (injectors etc)

to be honest and blunt this sounds like extremely inefficient and convoluted
using the numbers to reference to the same ones above:

1. we would need to read args of the entire prize pool incl opponents lastVs etc. and build entry structs from it as well as a whole prize pool struct as we need to know how many placements are set etc.
2. it would add to those structs, at that point basically the whole opponent processing is done entirely (possibly apart from the player teams) but then for your module to understand it we would have to deconstruct everything again as it can not read opponent structs but only the plain input
3. here the commons part will just ditch half the data again (factions and player teams) and then we would have to re-add it again via the injectors

so as a tldr: the /Custom has to do everything and then deconstruct it, call commons which destroys part of it again and readd stuff again via injectors
seems like waste on runtime etc. to me, theoretically with that setup we could skip the entire commons part of the input reading

so tbh atm i would lean towards not adding the standardized prize pools on sc/sc2 until full automation is added via commons
the sc/sc2 specific part of the automation is basically:
- the same race handling as with the manual input opponent handling
- additional limit handling (which could easily be done via /custom before hand and just set in the options)

(see https://liquipedia.net/starcraft2/index.php?title=Module:Hjpalpha/sandbox26&action=edit#mw-ce-l464 where i copied over the functions in dhe's prize pool that do the automation, as you can see sc/sc2 specific functions are only limit handling and race stuff)

Reply:
Let's take a call on this, I think we're talking around each other